### ISSUE TITLE
[Performance improvement] Add an option to use parent folder .gitignore during search

### ISSUE THREAD

**Issue by WildOgr** [microsoft/vscode#118229]
Currently VSCode uses ripgrep's --no-ignore-parent option. This is inconvenient and can significantly slow down search. E.g. if we consider following scenario:

Monorepo with following structure (various folders with same name excluded by parent .gitignore)

```
.gitignore
backend/
- project1
- project2
- etc.
frontend/
-project1
-etc.  
```

If user opens project1 folder in vscode, currently parent ignore settings will not be respected. It this can produce a lot of unexpected results and slows search performance.

https://github.com/microsoft/vscode/blob/f7ff53d7b06a2ce14ff503dc2290b487d9162429/src/vs/workbench/services/search/node/ripgrepFileSearch.ts#L62

https://github.com/microsoft/vscode/blob/f7ff53d7b06a2ce14ff503dc2290b487d9162429/src/vs/workbench/services/search/node/ripgrepTextSearchEngine.ts#L416

Proposal is to add an option to use search without --no-ignore-parent.

**Comment by samestep** [comment]
@roblourens Would you be open to a PR fixing this issue by adding a `useParentIgnoreFiles` option that disables the `--no-ignore-parent` flag (i.e. in the same way #59717 fixed #59364)?

**Comment by roblourens** [comment]
Yeah, I would take a setting to do that
